services/QuillLMS/engines/evidence/app/models/evidence/activity.rb

```diff
@@ -42,7 +46,9 @@ def set_parent_activity
         self.parent_activity = Evidence.parent_activity_class.find_or_create_by(
           name: title,
           activity_classification_id: Evidence.parent_activity_classification_class.evidence&.id
-        )
+        ) do |activity|
```

Comment:
For greater readability, maybe rename the block parameter `parent_activity` or `p_activity`  to distinguish from the evidence activity:

```) { |parent_activity| parent_activity.flags =  [LMS_ACTIVITY_DEFAULT_FLAG]```